src/libs/actions/User.ts

```diff
@@ -70,6 +70,14 @@ Onyx.connect({
     },
 });
 
+let allPolicies: OnyxCollection<Policy>;
+
+Onyx.connect({
```

Comment:
I'm not really a fan of subscribing to all policies in this file, can we subscribe to them in `ContactMethodDetailsPage` instead and use a selector to only get the policies owned by the user? then pass them to `setContactMethodAsDefault`

Reply:
Which is why I mentioned using a selector! You can select only the policy IDs owned by the user which is all we care about here and there won't be any re-renders as the ID will never change.

Reply:
>in this file we already subscribe all polices

I'm not sure I'm following, we are doing so as part of your PR. We are not subscribing to them in PolicyUtils either. We've started recommending to rely less on subscribing to Onyx in lib files as they cannot benefit from memoizing. Is there any issue I might be missing with subscribing in the component and using a selector? Like I mentioned above the selector should eliminate re-renders, and even if there was re-renders, I think it's fine as this component is isolated.

Reply:
>It's because the type we define for policies will be the returned type of selector, and it's not correct

I think it would be good if we could fix that type error to make it work with adding a selector and subscribe to policies in the component, there are high changes we'll need this in the future and it'll be nice if we get it fixed now. What do you think?

Reply:
thanks!

Reply:
Replied in the PR.

Reply:
@tienifr @youssef-lr Please have a look at [this comment](https://github.com/Expensify/react-native-onyx/pull/482/files#r1507479454).